### ISSUE TITLE
Listeners tab and listener type filter visibility

### ISSUE THREAD

**Issue by vitorwtavares** [camunda/camunda#25387]
### Description (required on creation)

This issue looks to address two things:

1) Currently the Listeners tab gets hidden when there are no listeners attached to a flow node or process. We now would like the Listeners tab to always be visible. When there are no listeners for that flow node or selected filter, a message should be displayed to reflect that.

2) The listener type filter dropdown should only be displayed only when the flow node (or flow node instance) selected is a User Task, otherwise it should be hidden (because all other types of elements that allow attaching a listener only allow execution listeners, and not Task listeners, thus making the filter useless).

### Acceptance Criteria (required on creation)

- [ ] Listeners tab should always be visible in the process instance page bottom panel
- [ ] When there are no listeners a message on the panel should reflect that
- [ ] When selecting a value on the listener type filter that has no listeners, the empty message should appear, but still allow another value on the dropdown to be selected
- [ ] When a User task is selected the listener type filter dropdown should be visible
- [ ] When any element other than a User task is selected the listener type filter dropdown should not be visible

### Links

- [Figma design](https://www.figma.com/design/zfOVIMNthZneqFiyUhWTP2/Listeners?node-id=2057-26289&node-type=section&m=dev)

```[tasklist]
### Pull Requests
```

**Comment by vitorwtavares** [comment]
Documenting for posterity:

In this issue we ran into a problem with re-renders on the Panel and Tab selection, more specifically the bottom panel on the Process Instance page.

The unwanted behavior here was that when a certain value was selected on the listeners type filter this caused the Listeners tab to be deselected and for the default tab to be selected, as well as resetting the value selected on the filter once you navigated to the Listeners tab again.

After attempting to debug this in every way possible related to the most likely suspects (useEffect dependencies as well as changes in the store that could be causing data to change and in turn cause re-renders), @pedesen pointed out to me it was likely something a little bit more hidden: a key added to the `TabView` component. We had added this key a while prior and made it dynamic by changing it according to the existence (or not) of listeners attached to a certain element (flow node or flow node instance) in order for it to force a re-render and reset to the default Tab once there were no listeners to be shown and the Listeners tab got hidden.

With these new changes, the idea is to always show the Listeners Tab, so we did not need this behavior in this specific scenario anymore, but this kind of re-render manipulation via key was not very clear nor was it documented in this instance.

We are now using a variation of this key manipulation to be able to purposely reset the tab selection once the selected element changes.

TLDR: A custom dynamic key introduced earlier in the development of the Listeners tab was causing the now unexpected re-renders, this was not very clear, easy to debug or documented everywhere but eventually we found the cause and solved the issue.

**Comment by vitorwtavares** [comment]
🚫 Issue marked blocked in Jan 15th:

We currently have 2 problems on E2E tests (which might be related):

1. On CI the tests are failing because no listeners show up even when a user task gets completed (should have an end execution listener as well as a "complete" task listener. In order to reproduce this locally we ran into problem number 2 (see below).
2. On local environment when running listeners E2E tests the metadata for our User task is missing the `userTaskKey`, which makes it impossible for the test to make the request to the Rest API to complete that user task and in turn for the listeners to show up in the UI. This missing `userTaskKey` appears to be a backend issue (might be related to exporter changes, as this was previously working a couple of weeks ago and the bug coincides with the transition period for importer/exporter on the backend, but this is being investigated by @sdorokhova) and thus is a blocker to reproducing and fixing the E2E test issue we get on the CI. The issue was reproducible in some instances in the main branch as well, but seems to occur inconsistently.

**Comment by vitorwtavares** [comment]
✅ Blocker resolved on Feb 26th:

It is hard to say for certain what the root cause of the blockers were (we had several, some of them got fixed between the last update and now, others just "disappeared"). When originally working on this issue, there were two big restructuring projects going on simultaneously with high impact (and therefore likelihood for bugs) on the codebase: change from importers to exporters as well as API migration to V2. It is most likely some of the changes from those project impacted Operate's E2E tests and tests setup.

After a while on blocked with no progress, I revisited the issue and the errors I was getting changed. After a bit of investigation I realized the wrong port was being used for the frontend E2E tests setup (most likely due to changes in the `make` commands that were originated from changes in how the exporter works) in the `ZEEBE_REST_GATEWAY_ADDRESS` variable, I changed the port and the C8 JS SDK started working again - in this case successfully completing a user task in one of the E2E tests.

After that there were code adjustments made in the Frontend E2E tests themselves and it eventually passed both on the local environment and on the CI when the new commits got pushed to the remote branch.

**Comment by vitorwtavares** [comment]
@korthout I believe this is already [completed here](https://github.com/camunda/camunda/pull/25969), is there something else to do? If nothing changed I think we can close this issue.
